### PR TITLE
Add puppetdb manage dbserver

### PR DESCRIPTION
**Pull request by chrisowensboston**
Added manage_dbserver param to profile/puppetdb, to make it consistent with profile/master.

### CODE REVIEW

**Comment by rendhalver** [comment]
Can you also remove the Gemfile.lock changes from this please.

**Comment by chrisowensboston** [comment]
I probably can’t get to that until tomorrow. Stand by.

File: spec/classes/puppet_profile_puppetdb_spec.rb [hunk 1]
```diff
@@ -67,6 +67,7 @@
             :disable_ssl        => false,
             :listen_address     => '127.0.0.1',
             :ssl_listen_address => '0.0.0.0',
+            :manage_dbserver    => true,
```
Comment (rendhalver) [inline spec/classes/puppet_profile_puppetdb_spec.rb]: Can you also add a test for when this is not set.

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,7 +23,6 @@ In .fixtures.yml, changed puppetlabs/concat version requirement from '1.2.0' to
 in Gemfile, forced json_pure to be < 2.0.1 when ruby < 2.0 @chrisowensboston
 in .travis.yml, added a line for rvm 2.2.0, puppet gem 4.7.0, facter gem 2.4.6, hiera gem 3.2.1 @chrisowensboston
 
-
```
Comment (rendhalver) [inline CHANGELOG.md]: You can remove your changes to this file too.

File: spec/classes/puppet_profile_puppetdb_spec.rb (hunk 1 above)
Comment (chrisowensboston) [inline spec/classes/puppet_profile_puppetdb_spec.rb]: Adding that test surfaced a bug in the puppetlabs/postgresql module: https://tickets.puppetlabs.com/browse/MODULES-4054.

Comment (rendhalver) [inline spec/classes/puppet_profile_puppetdb_spec.rb]: Ouch.
Maybe we need to looks at later versions of the module.
I will have a look later and se what we can do about this.

**Comment by chrisowensboston** [comment]
The problem is probably theoretical… it occurs in testing, because declaring puppet::profile::puppetdb with manage_dbserver false, invokes the puppetdb class with manage_dbserver false, which declares an instance of postgresql::server::db but (because it’s not trying to manage the server) does not declare postgresql::server. 

In actual deployments, the the likely reason for setting manage_dbserver to false would be because you’re managing the Postgresql server out of some other catalog entry, i.e. postgresql::server is going to be declared on that host somewhere, somehow. The exception would be a host that has a postgresql server on it that is not under the control of puppet. 

I’m not sure what the right answer would be. I don’t love the whole postgresql::globals pattern, but perhaps the right thing would be for postgresql::server::db and friends to include postgresql::globals and get things like character encoding from that class?

**Comment by rendhalver** [comment]
Yeah that makes sense.
Annoying but it explains what the issue is.
I think I did run into something similar when I was trying to spec test puppetdb in this module.
Maybe we should just leave it off the tests and trust that puppet is testing things properly.

File: spec/classes/puppet_profile_puppetdb_spec.rb (hunk 1 above)
Comment (rendhalver) [inline spec/classes/puppet_profile_puppetdb_spec.rb]: Given discussion in main thread lets leave this for now.

Comment (chrisowensboston) [inline spec/classes/puppet_profile_puppetdb_spec.rb]: Where is discussion?

**Comment by chrisowensboston** [comment]
I can't make sense of the pull request chronology... are you still waiting for changes from me?

**Comment by rendhalver** [comment]
@chrisowensboston Please rebase and remove your changes to the CHANGELOG.md

**Comment by chrisowensboston** [comment]
Sorry, been tied up. Will look at this in the AM.

**Comment by chrisowensboston** [comment]
Hi Pete (sorry for muddying the waters by replying here but I don’t have an e-mail address for you).

 

Could you help me verify my own sanity by trying this and letting me know what you see? 

Very oddly, puppet agent v4.9 and 4.10 (at least) seem to read the puppet.conf file differently depending upon whether run as a service or run manually. 

https://tickets.puppetlabs.com/browse/PUP-7434

 

Thanks much,

 

/c

**Comment by rendhalver** [comment]
Interesting.
That's going to break a few setups.
I have been seeing some interesting things on some recent servers that could be explained by that.
Let me do some investigation and get back to you.

**Comment by rendhalver** [comment]
Yep it looks like that is what's happening.
I migrated a few of my servers to a new puppet server recently.
The old server listens on puppet.$::domain. The new servers uses a different address.
The servers I migrated are contacting the old puppet server and getting auth denied.

**Comment by rendhalver** [comment]
@chrisowensboston If you are happy with this I am going to merge it.

**Comment by rendhalver** [comment]
Actually can you please rebase @chrisowensboston ?

**Comment by chrisowensboston** [comment]
Will do


Sent from handheld; pls excuse terseness...

-------- Original message --------

**Comment by calmenergy** [comment]
It's possible that this branch is now munted due to me having made a mess of the rebase.

**Comment by rendhalver** [comment]
Yes it looks like it.
It's best to just setup an upstream remote and sync up your local production branch with that and then do a manual rebase.
I don't use the sync buttons in GitHub at all.
This will show you how to sync a fork: https://help.github.com/articles/syncing-a-fork/

**Comment by calmenergy** [comment]
At this point, my end is such a mess of commits that I’m going to create a new fork of your ‘production’ branch, manually make my changes to that branch, commit all in one, and then do a new PR.

**Comment by rendhalver** [comment]
git cherry-pick is a good way to copy commits to a new branch.

**Comment by calmenergy** [comment]
See PR #144

**Pull request closed without merging**